lms/djangoapps/bulk_email/models.py

```diff
@@ -107,7 +107,7 @@ def get_users(self, course_id, user_id=None):
         staff_qset = CourseStaffRole(course_id).users_with_role()
         instructor_qset = CourseInstructorRole(course_id).users_with_role()
         staff_instructor_qset = (staff_qset | instructor_qset)
-        enrollment_qset = User.objects.filter(
+        enrollment_query = models.Q(
```

Comment:
We may do this by using **kwargs. Like we can do as 
```python
 query_kwargs = { is_active:True, courseenrollment__course_id:course_id, courseenrollment__is_active:True} 
```
and then 
``` python 
enrollment_qset =  User.objects.filter(**query_kwargs) 
```
this way we can avoid redundant calls to `User.objects.filter` at line 123 and 129.
At line 132 `query_kwargs` can be updated, 
```python
 query_kwargs.update({courseenrollment__mode:self.coursemodetarget.track.mode_slug})
```
Here again 
``` python 
enrollment_qset =  User.objects.filter(**query_kwargs) 
```
By doing this we will be saving cost of Q objects also.